src/core/k8.mjs

```diff
@@ -197,14 +197,14 @@ export class K8 {
 
   /**
    * Get pods by labels
+   * @param namespace the namespace of the pods
    * @param labels list of labels
    * @return {Promise<Array<V1Pod>>}
    */
-  async getPodsByLabel (labels = []) {
-    const ns = this._getNamespace()
+  async getPodsByLabel (namespace, labels = []) {
```

Comment:
Not sure we need to pass namespace in the k8 function since k8 instances picks it from configManager. So as long as configManager is up-to-date, it should work.

Reply:
same response